Source/Engine/PackageBuilder/PatternLinker.cs

```diff
@@ -90,61 +64,30 @@ protected internal override Syntax VisitRequiredPackage(RequiredPackageSyntax no
             return result;
         }
 
-        protected internal override Syntax VisitSearchTarget(SearchTargetSyntax node)
-        {
-            SearchTargetSyntax result = node;
-            if (node.IsNamespaceWithWildcard())
-            {
-                string ns = node.SearchTarget.TrimEnd('*', '.');
-                var targetReferences = new List<Syntax>();
-                foreach (PatternSyntax p in fPatternByName.Values.Where(x => x.IsSearchTarget && x.Namespace == ns))
-                {
-                    PatternReferenceSyntax r = Syntax.PatternReference(p.FullName);
-                    var rc = new PatternReferenceInContext()
-                    {
-                        PatternReference = r,
-                        PatternContext = null
-                    };
-                    fPatternReferences.Add(rc);
-                    targetReferences.Add(r);
-                }
-                result = new NamespaceSearchTargetSyntax(ns, targetReferences);
-            }
-            else
-            {
-                if (fPatternByName.TryGetValue(node.SearchTarget, out PatternSyntax pattern))
-                {
-                    PatternReferenceSyntax targetReference = Syntax.PatternReference(pattern);
-                    result = new PatternSearchTargetSyntax(pattern.FullName, targetReference);
-                }
-                else
-                    throw SyntaxError(TextResource.SearchTargetIsUndefinedPattern, node.SearchTarget);
-            }
-            return result;
-        }
-
         protected internal override Syntax VisitPatternSearchTarget(PatternSearchTargetSyntax node)
         {
-            var rc = new PatternReferenceInContext()
-            {
-                PatternReference = (PatternReferenceSyntax)node.PatternReference,
-                PatternContext = null
-            };
-            fPatternReferences.Add(rc);
+            if (fPatternByName.TryGetValue(node.SearchTarget, out PatternSyntax pattern))
+                node.PatternReference.ReferencedPattern = pattern;
+            else
+                throw SyntaxError(TextResource.SearchTargetIsUndefinedPattern, node.SearchTarget);
             return node;
         }
 
         protected internal override Syntax VisitNamespaceSearchTarget(NamespaceSearchTargetSyntax node)
         {
-            foreach (PatternReferenceSyntax r in node.PatternReferences)
+            var targetReferences = new List<Syntax>();
+            foreach (PatternSyntax p in fPatternByName.Values.Where(x => x.IsSearchTarget && x.Namespace == node.Namespace))
```

Comment:
There is no guarantee that at the moment of VisitNamespaceSearchTarget execution the fPatternByName contains all patterns from all namespaces.